var/spack/repos/builtin/packages/hip-examples/package.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import os
+
+from spack.package import *
+
+
+class HipExamples(Package):
+    """Examples for HIP"""
+
+    homepage = "https://github.com/ROCm-Developer-Tools/HIP-Examples/"
+    git = "https://github.com/ROCm-Developer-Tools/HIP-Examples.git"
+    url = "https://github.com/ROCm-Developer-Tools/HIP-Examples/archive/rocm-5.4.3.tar.gz"
+    tags = ["rocm"]
+
+    maintainers("srekolam", "renjithravindrankannath", "afzpatel")
+
+    version("master", branch="master")
+
+    version("5.4.3", sha256="053b8b7892e2929e3f90bd978d8bb1c9801e4803eadd7d97fc6692ce60af1d47")
+
+    patch("0001-add-inc-and-lib-paths-to-openmp-helloworld.patch")
+    patch("0002-add-fpic-compile-to-add4.patch")
+    depends_on("hip")
+    depends_on("rocm-openmp-extras")
+
+    def install(self, spec, prefix):
+        stage = os.getcwd()
+        os.environ["ROCM_OPENMP_EXTRAS_DIR"] = self.spec["rocm-openmp-extras"].prefix
+        os.environ["LD_LIBRARY_PATH"] = self.spec["rocm-openmp-extras"].prefix.lib
+        for root, dirs, files in os.walk(stage):
+            if "Makefile" in files:
+                with working_dir(root, create=True):
+                    make()
+                    # itterate through the files in dir to find the newly built binary
+                    for file in os.listdir("."):
+                        if file not in files and os.path.isfile(file) and os.access(file, os.X_OK):
+                            install(file, join_path(prefix, file))
+                            if file == "RecursiveGaussian":
+                                install(
+                                    "RecursiveGaussian_Input.bmp",
+                                    join_path(prefix, "RecursiveGaussian_Input.bmp"),
+                                )
+
+    def test(self):
+        os.environ["LD_LIBRARY_PATH"] = self.spec["rocm-openmp-extras"].prefix.lib
+        test_dir = self.spec["hip-examples"].prefix
```

Comment:
Tests should never be built/run from a prefix (or install) directory.  This can create problems(e.g., permissions) for facilities, read-only file system installations, and shared spack instances.

A test stage directory is automatically created for stand-alone tests and should be used for such purposes.  If the executables cannot be run from the stage directory then they should be copied prior to running them.

Reply:
https://spack.readthedocs.io/en/latest/packaging_guide.html#adding-stand-alone-tests
In the test_example function given in the documentation, isn't example being run from the prefix directory or am I misunderstanding something?

Reply:
Not the prefix directory.  That is problematic for some sites/facilities/use cases.

Spack sets up a test stage directory (https://spack.readthedocs.io/en/latest/packaging_guide.html#configuring-the-test-stage-directory) when `spack test run` is executed.  Tests are then run out of it by default.